src/pybind/mgr/volumes/fs/async_job.py

```diff
@@ -263,7 +324,9 @@ def _cancel_jobs(self, volname):
                 log.debug("waiting for {0} in-progress jobs for volume '{1}' to "
                           "cancel".format(len(self.jobs[volname]), volname))
                 self.cancel_cv.wait()
-            self.jobs.pop(volname)
+
+            if update_queue:
```

Comment:
Have you checked if it suffices to remove the volume from `self.jobs`? I think it should be fine to remove. Please double check.

Reply:
What about this comment @rishabh-d-dave ?